### PR TITLE
Use GetIPFromRequest helper func everywhere

### PR DESCRIPTION
**Pull request by mvdan**
We had semi-working copies all over the place. They were incomplete and
buggy like the old GetIPFromRequest, like how they missed ipv6 support.

Use the common helper func, which is also unit tested.

### CODE REVIEW

**Comment by mvdan** [comment]
@lonelycode note that this touches the log call I modified recently to use `r.RemoteAddr`. I don't see how that log call could ever fire or why it's useful, as it's from an error to get an IP address that we never use.

File: handler_error.go [hunk 1]
```diff
@@ -212,32 +209,6 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 		w.Header().Add("Connection", "close")
 	}
 
-	var obfuscated string
```
Comment (buger) [inline handler_error.go]: It seems like this part was completely removed, without being replaced by `GetIPFromRequest`. Even if `clientIP` not used anywhere, it still miss error logging, in case if `RemoteAddr` has issues.

**Comment by lonelycode** [comment]
> note that this touches the log call I modified recently to use r.RemoteAddr. I don't see how that log call could ever fire 

When could it fire: When `SplitHostPort()` fails, which admittedly is unlikely, but it would be abnormal to fail, and in the interest of exposing abnormal behaviour, I'd like an error message there to cover the scenario. (See all the error scenarios in the source: https://golang.org/src/net/ipsock.go?s=3572:3638#L104)

> or why it's useful, as it's from an error to get an IP address that we never use.

The IP address *is* used, it is used to modify the `X-Forward-For` header, which informs the upstream server  of the proxy path that this request has gone through, if someone were messing with the chain, it might be useful to inform the user...

**Comment by mvdan** [comment]
> When SplitHostPort() fails, which admittedly is unlikely, but it would be abnormal to fail, and in the interest of exposing abnormal behaviour, I'd like an error message there to cover the scenario.

Let me rephrase this - what we pass is `r.RemoteAddr`, which `net/http` says will be a valid addr+port in the very same format. Unless we have messed with `RemoteAddr` (hopefully not), this would only fire if Go's standard library is buggy.

> The IP address is used, it is used to modify the X-Forward-For header

Sorry, where? We only get the header - we don't set it nor modify it. Really, all we were doing there is the old version of `GetIPFromRequest`.

File: handler_error.go (hunk 1 above)
Comment (mvdan) [inline handler_error.go]: See my comment above and the discussion below.

**Comment by lonelycode** [comment]
Sweet Christmas. Just leave it alone please.

**Comment by mvdan** [comment]
As you wish.

**Comment by mvdan** [comment]
@buger rebased, PTAL
